packages/devtools_app/lib/src/shared/diagnostics/generic_instance_reference.dart

```diff
@@ -63,7 +63,9 @@ class ObjectReferences extends GenericInstanceRef {
     required IsolateRef super.isolateRef,
     required super.value,
     required HeapObjectSelection super.heapSelection,
-  });
+  }) {
+    if (refNodeType.isLive) assert(value != null);
```

Comment:
Nit: `assert(!refNodeType.isLive || value != null)`

Reply:
Current option looks more readable for me.